model/riscv_insts_vext_fp.sail

```diff
@@ -0,0 +1,1359 @@
+  mask        : vector('n, dec, bool)     = undefined;
+
+  (result, mask) = init_masked_result(num_elem, SEW, LMUL_pow, vd_val, vm_val);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if mask[i] then {
+      result[i] = match vfunary1 {
+        FVV_VSQRT      => {
+                            let (fflags, elem) : (bits_fflags, bits('m)) = match 'm {
+                              16  => riscv_f16Sqrt(rm_3b, vs2_val[i]),
+                              32  => riscv_f32Sqrt(rm_3b, vs2_val[i]),
+                              64  => riscv_f64Sqrt(rm_3b, vs2_val[i])
+                            };
+                            write_fflags(fflags);
+                            elem
+                          },
+        FVV_VRSQRT7    => {
+                            let (fflags, elem) : (bits_fflags, bits('m)) = match 'm {
+                              16  => riscv_f16Rsqrte7(rm_3b, vs2_val[i]),
+                              32  => riscv_f32Rsqrte7(rm_3b, vs2_val[i]),
+                              64  => riscv_f64Rsqrte7(rm_3b, vs2_val[i])
+                            };
+                            accrue_fflags(fflags);
+                            elem
+                          },
+        FVV_VREC7      => {
+                            let (fflags, elem) : (bits_fflags, bits('m)) = match 'm {
+                              16  => riscv_f16Recip7(rm_3b, vs2_val[i]),
+                              32  => riscv_f32Recip7(rm_3b, vs2_val[i]),
+                              64  => riscv_f64Recip7(rm_3b, vs2_val[i])
+                            };
+                            accrue_fflags(fflags);
+                            elem
+                          },
+        FVV_VCLASS     => fp_class(vs2_val[i])
+      }
+    }
+  };
+
+  write_vreg(num_elem, SEW, LMUL_pow, vd, result);
+  vstart = zeros();
+  RETIRE_SUCCESS
+}
+
+mapping vfunary1_mnemonic : vfunary1 <-> string = {
+  FVV_VSQRT       <-> "vfsqrt.v",
+  FVV_VRSQRT7     <-> "vfrsqrt7.v",
+  FVV_VREC7       <-> "vfrec7.v",
+  FVV_VCLASS      <-> "vfclass.v"
+}
+
+mapping clause assembly = VFUNARY1(vm, vs2, vfunary1, vd)
+  <-> vfunary1_mnemonic(vfunary1) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ maybe_vmask(vm)
+
+/* ****************************** OPFVV (VWFUNARY0) ****************************** */
+union clause ast = VFMVFS : (regidx, regidx)
+
+mapping clause encdec = VFMVFS(vs2, rd) if haveRVV()
+  <-> 0b010000 @ 0b1 @ vs2 @ 0b00000 @ 0b001 @ rd @ 0b1010111 if haveRVV()
+
+function clause execute(VFMVFS(vs2, rd)) = {
+  let rm_3b    = fcsr.FRM();
+  let SEW      = get_sew();
+  let num_elem = get_num_elem(0, SEW);
+
+  if illegal_fp_vd_unmasked(SEW, rm_3b) | SEW > sizeof(flen)
+  then { handle_illegal(); return RETIRE_FAIL };
+  assert(num_elem > 0 & SEW != 8);
+
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, 0, vs2);
+  F(rd) = nan_box_vector(vs2_val[0]);
```

Comment:
Reuse the existing F_* functions like get_scalar_fp so as to not have explicit NaN-boxing here

Reply:
This one is modified as suggested.